nemo/collections/llm/tools/autotuner/README.md

```diff
@@ -0,0 +1,421 @@
+# AutoTuner
+
+AutoTuner is a fully automated, predictive configuration builder and orchestrated tool that achieves maximum model throughput for LLM Training.
+
+## Overview
+
+AutoTuner extends NeMo's Auto Configurator with a fully automated pipeline that searches for hyperparameters achieving maximum training throughput while automatically preventing CUDA OOM errors and providing comprehensive performance & cost analysis for Large Language Models. Unlike research-only tools, AutoTuner runs directly on your GPU infrastructure, making it ready for production deployment and real-world training optimization.
```

Comment:
I would say "automatically detecting potential model configurations that will throw CUDA OOM errors..." as it doesn't actually prevent CUDA OOM errors themselves, but skips jobs that will very likely trigger them.